src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -176,22 +177,32 @@ public WebSocketsTransport(WebSocketOptions options, Channel<byte[]> application
                     {
                         try
                         {
-                            _logger.SendPayload(_connectionId, buffer.Length);
+                            _logger.SendPayload(_connection.ConnectionId, buffer.Length);
+
+                            var webSocketMessageType = _options.WebSocketMessageType;
+                            var transferModeFeature = _connection.Features.Get<ITransferModeFeature>();
+                            if (transferModeFeature != null)
```

Comment:
Correct. We set it after the transport is created bug before the first message is sent. It should not change after that.